### PR TITLE
Add IPv6 support to Kiwix Server

### LINKED ISSUE #545: kiwix-serve does not support IPv6
```
root@hertz-ms ~ # kiwix-serve kiwix/* -i 2a0f:9400:7a0a::5
Ip address 2a0f:9400:7a0a::5  is not a valid ip address

root@hertz-ms ~ # kiwix-serve kiwix/* -i "[2a0f:9400:7a0a::5]"
Ip address [2a0f:9400:7a0a::5]  is not a valid ip address
```

### PR DESCRIPTION
**Pull request by aryanA101a**
Fixes #545 

depends on https://github.com/kiwix/libkiwix/pull/1074

### CODE REVIEW

File: src/server/kiwix-serve.cpp [hunk 1]
```diff
@@ -307,6 +309,9 @@ int main(int argc, char** argv)
         case 's':
           searchLimit = atoi(optarg);
           break;
+        case '6':
+          ipv6Flag = true;
+          break;
```
Comment (veloman-yunkan) [inline src/server/kiwix-serve.cpp:315]: Can't the type of the address be detected automatically?

File: docs/kiwix-serve.rst [hunk 2]
```diff
@@ -51,7 +51,11 @@ Options
 .. option:: -i ADDR, --address=ADDR
 
   Listen only on this IP address. By default the server listens on all
-  available IP addresses.
+  available IP addresses. Alternatively, you can use special values to define which types of connections to accept:
+
+  all : Accept connections from all IP addresses (IPv4 and IPv6).
+  ipv4 : Accept connections from all IPv4 addresses.
+  ipv6 : Accept connections from all IPv6 addresses.
```
Comment (veloman-yunkan) [inline docs/kiwix-serve.rst:58]: "Accept connections from an IP address" is very different from "Listen for connections on an IP address"

File: src/man/kiwix-serve.1 [hunk 3]
```diff
@@ -28,7 +28,13 @@ By default, kiwix-serve expects a list of ZIM files as command line arguments. P
...
 .TP
 \fB-i ADDR, --address=ADDR\fR
-Listen only on this IP address. By default, the server listens on all available IP addresses.
+Listen only on this IP address. By default, the server listens on all available IP addresses. Alternatively, you can use special values to define which types of connections to accept:
+
+all : Accept connections from all IP addresses (IPv4 and IPv6).
+.br
+ipv4 : Accept connections from all IPv4 addresses.
+.br
+ipv6 : Accept connections from all IPv6 addresses.
```
Comment (veloman-yunkan) [inline src/man/kiwix-serve.1:37]: Same here

File: src/server/kiwix-serve.cpp [hunk 4]
```diff
@@ -63,7 +63,7 @@ void usage()
             << "\t-h, --help\t\tPrint this help" << std::endl << std::endl
             << "\t-a, --attachToProcess\tExit if given process id is not running anymore" << std::endl
             << "\t-d, --daemon\t\tDetach the HTTP server daemon from the main process" << std::endl
-            << "\t-i, --address\t\tListen only on this ip address, all available ones otherwise" << std::endl
+            << "\t-i, --address\t\tListen only on this ip address, all available ones otherwise, or specify special values to accept connections from ipv4/ipv6/all devices" << std::endl
```
Comment (veloman-yunkan) [inline src/server/kiwix-serve.cpp:66]: Change to

Listen only on the specified IP address. Specify 'ipv4', 'ipv6' or 'all' to listen on all IPv4, IPv6 or both types of addresses, respectively (default: all).

**Review by veloman-yunkan — CHANGES_REQUESTED** [review]

**Comment by kelson42** [comment]
@aryanA101a Now that https://github.com/kiwix/libkiwix/pull/1074 is (almost) ready, we should focus on this PR. Is this PR ready for a new review? Could you please give a specific feedback to each comment of the reviewer?

**Comment by kelson42** [comment]
@aryanA101a Can we complete this PR?

**Comment by aryanA101a** [comment]
I'm not sure why the build is failing. Is it due to `dpkg-buildpackage` does something different?
https://github.com/kiwix/kiwix-tools/actions/runs/9188369878/job/25571181092?pr=673#step:9:432
Perhaps it is not linking with libkiwix.so

**Comment by kelson42** [comment]
@aryanA101a This is something out of the scope of this PR. I'm more concerned about the fact that other parts of the CI (like compilation on Linux) is not triggered at all!

**Comment by mgautierfr** [comment]
> I'm more concerned about the fact that other parts of the CI (like compilation on Linux) is not triggered at all!

Should be fixed by #679

File: docs/kiwix-serve.rst [hunk 5]
```diff
@@ -51,7 +51,11 @@ Options
 .. option:: -i ADDR, --address=ADDR
 
   Listen only on this IP address. By default the server listens on all
-  available IP addresses.
+  available IP addresses. Alternatively, you can use special values to define which types of connections to accept:
+
+  all : Listen for connections from all IP addresses (IPv4 and IPv6).
+  ipv4 : Listen for connections from all IPv4 addresses.
+  ipv6 : Listen for connections from all IPv6 addresses.
```
Comment (veloman-yunkan) [inline docs/kiwix-serve.rst]: 1. from → on
2. Make it a bulleted list (start each line with a `- `). Build (via `ninja doc`) and check the documentation to make sure that it looks nice.

File: src/man/kiwix-serve.1 [hunk 6]
```diff
@@ -28,7 +28,13 @@ By default, kiwix-serve expects a list of ZIM files as command line arguments. P
...
 .TP
 \fB-i ADDR, --address=ADDR\fR
-Listen only on this IP address. By default, the server listens on all available IP addresses.
+Listen only on this IP address. By default, the server listens on all available IP addresses. Alternatively, you can use special values to define which types of connections to accept:
+
+all : Listen for connections from all IP addresses (IPv4 and IPv6).
+.br
+ipv4 : Listen for connections from all IPv4 addresses.
+.br
+ipv6 : Listen for connections from all IPv6 addresses.
```
Comment (veloman-yunkan) [inline src/man/kiwix-serve.1]: from → on

**Comment by veloman-yunkan** [comment]
Since I actually performed the proposed changes locally (in order to find out the proper formatting of documentation) I also committed them. If you are fine with them, please rebase&fix-up and we will merge this PR.

**Comment by kelson42** [comment]
@aryanA101a Please squash everything in one commit

**Pull request merged**